development/auto-changelog.sh

```diff
@@ -1,26 +1,63 @@
-#! /bin/bash
-# update tags
+#!/bin/env bash
+
+set -e
+set -u
+set -o pipefail
+
+readonly URL='https://github.com/MetaMask/metamask-extension'
+
 git fetch --tags
-# get origin
-URL='https://github.com/MetaMask/metamask-extension'
-# get git logs from last tag until HEAD, pretty by 'subject::body' filtered by grep for PRs made with Github squash merge or Github regular merge
-LOG=$(git log $(git describe --tags $(git rev-list --tags --max-count=1))..HEAD --pretty="%s::%b" --reverse --grep="Merge pull request #" --grep="(#");
-while read -r line; do
-    # get git log subject
-    SUBJECT=$(echo $line | sed -E 's/(.*):{2}(.*)/\1/')
-    # get git log PR id, PR made with Github squash merge or Github regular merge
-    PR=$(echo $SUBJECT | sed 's/^.*(#\([^&]*\)).*/\1/' | sed 's/^.*#\([^&]*\) from.*/\1/')
-    # if PR made with Github squash merge, subject is the body
-    if [ -z "$(echo $line | sed -E 's/(.*):{2}(.*)/\2/')" ]; then
-        BODY=$(echo $SUBJECT | sed "s/(#$PR)//g"); else
-        BODY=$(echo $line | sed -E 's/(.*):{2}(.*)/\2/') 
+
+most_recent_tag="$(git describe --tags "$(git rev-list --tags --max-count=1)")"
+
+readarray -t commits <<< "$(git rev-list "${most_recent_tag}"..HEAD)"
+
+for commit in "${commits[@]}"
+do
+    subject="$(git show -s --format="%s" "$commit")"
+
+    # Parse a commit for the PR ID and a description. Three types of commits
+    # are handled: Squash & Merge, Merge, and normal commits.
+    # Squash & Merge: the commit subject is parsed as `<description> (#<PR ID>)`
+    # Merge: the PR ID is parsed from the git subject (which is of the form `Merge pull request
+    #   #<PR ID> from <branch>`, and the description is assumed to be the first line of the body.
+    #   If no body is found, the description is set to the commit subject
+    # Normal commits: The commit subject is the description, and the PR ID is omitted.
+    if grep -E -q '\(#[[:digit:]]+\)|#[[:digit:]]+\sfrom' <<< "$subject"
+    then
+        # shellcheck disable=SC2001
+        pr="$(printf "%s" "$subject" | sed 's/^.*#\([[:digit:]]\+\).*$/\1/')"
+        prefix="[#$pr]($URL/pull/$pr): "
+
+        first_line_of_body="$(git show -s --format="%b" "$commit" | head -n 1 | tr -d '\r')"
+
+        if grep -E -q '\(#[[:digit:]]+\)' || [[ -z "$first_line_of_body" ]]
```

Comment:
Is this invocation of `grep` missing an input?

Reply:
Oops, yes it is, there should be a `$subject` there.